signal.go

```diff
@@ -20,9 +20,10 @@ var shutdownRequestChannel = make(chan struct{})
 var interruptSignals = []os.Signal{os.Interrupt}
 
 // shutdownListener listens for OS Signals such as SIGINT (Ctrl+C) and shutdown
-// requests from shutdownRequestChannel.  It returns a context that is canceled
-// when either signal is received.
-func shutdownListener() context.Context {
+// requests from shutdownRequestChannel.  It returns a context and its
+// cancellation function. The context is cancelled when either signal
+// is received.
+func shutdownListener() (context.Context, context.CancelFunc) {
```

Comment:
This change needs to be reverted.  It is unused and not a safe way to about it anyway since its use would be result in the signal goroutine never terminating.